src/qibolab/pulses.py

```diff
@@ -681,6 +682,7 @@ def __init__(
         channel,
         type=PulseType.DRIVE,
         qubit=0,
+        coupler=None,
```

Comment:
Do we need both a `qubit` and a `coupler`? Can't we just pass the `coupler` as the `qubit`? The pulse can only act on a single thing anyway, either a coupler or a qubit, not both together.

Reply:
It was mainly to avoid having to write checks like `if isinstance(pulse.qubit, Coupler)` or things like that. It would be a bit weird.

Reply:
If this information is already available in that way it's not weird, and it would be even convenient wrt duplicating it in multiple tracks.